brvs-core/src/main/java/iroha/validation/utils/ValidationUtils.java

```diff
@@ -74,6 +79,25 @@
       )
   );
 
+  static ToriiResponse sendWithLastResponseWaiting(
+      IrohaAPI irohaAPI,
+      Transaction transaction) {
+    return irohaAPI.transaction(
+        transaction,
+        subscriptionStrategy
+    ).takeUntil(Observable.timer(timeoutForIrohaStatus, TimeUnit.MINUTES))
```

Comment:
As I understood from [documentation](http://reactivex.io/documentation/operators/takeuntil.html), that not affect original subscription. Seems it stays still open after. Are you sure that we won't exceed grpc connection limit?